src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/config/VisitSchedulerExceptionHandler.kt

```diff
@@ -170,6 +171,20 @@ class VisitSchedulerExceptionHandler {
       )
   }
 
+  @ExceptionHandler(PublishEventException::class)
+  fun handlePublishEventException(e: PublishEventException): ResponseEntity<ErrorResponse?>? {
+    log.debug("Publish event exception caught: {}", e.message)
+    return ResponseEntity
+      .status(HttpStatus.INTERNAL_SERVER_ERROR)
+      .body(
+        ErrorResponse(
+          status = HttpStatus.INTERNAL_SERVER_ERROR,
+          userMessage = "Failed to publish event: ${e.cause?.message}",
+          developerMessage = e.message
```

Comment:
should exclude this if production or perhaps if production exception class name

Reply:
Same again, need a ticket to review the error handling

Reply:
sure